x-pack/plugins/security_solution/public/exceptions/api/list_api.ts

```diff
@@ -80,12 +81,13 @@ export const updateList = async ({ list, http }: UpdateExceptionList) => {
 
 export const unlinkListFromRules = async ({ rules, listId }: UnlinkListFromRules) => {
   try {
+    if (!rules.length) return;
     const abortCtrl = new AbortController();
-    rules.map((rule) => {
```

Comment:
I don't know how I missed this when reviewing the code! My apologies!

I think you also could have wrapped the `rules.map` in a `await Promise.all(rules.map(...` but if the preference here is to use the `asyncForEach` that is sufficient too since we are not returning the result of the `patchRule` + `map`.

Reply:
I also tried the `await rules.map((rule) => {` but it didn't work :(